lib/elixir/pages/Guards.md

```diff
@@ -209,4 +218,4 @@ def my_function(number) when is_even(number) do
 end
 ```
 
-While it's possible to create custom guards with macros, it's recommended to define them using `defguard` and `defguardp` which perform additional compile-time checks.
+While it's possible to create custom guards with macros, it's recommended to define them using `defguard/1` and `defguardp/1` which perform additional compile-time checks and optimizations.
```

Comment:
Do they perform optimizations?

Reply:
Yes, when they are used outside guards

Reply:
I wouldn't call that optimizations, it is more that it keeps evaluation semantics, so I would remove that note.

Reply:
https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/kernel/utils.ex#L159

Reply:
Unquoting once per argument is about preserving evaluation semantics. Maybe I am missing something so can you link to the specific code or docs?

Reply:
I submitted that link before reading your comment

Reply:
Just to clarify, it can be an optimization, but it is primarily to preserve evaluation semantics, specifically that each argument is evaluated once like expected.

Reply:
Agreed with @ericmj.

Reply:
reverted. thank you for the discussion